donut/modules/permissions/helpers.py

```diff
@@ -0,0 +1,24 @@
+import flask
+from donut.modules.groups import helpers as groups
+
+
+def has_permission(user_id, permission_id):
+    ''' 
+    Returns True if [user_id] holds a position that directly 
+    or indirectly (through a position relation) grants 
+    them [permission_id]. Otherwise returns False.
+    '''
+    if not (isinstance(user_id, int) and isinstance(permission_id, int)):
+        return False
+    # get all position id's with this permission
+    query = '''SELECT pos_id FROM position_permissions WHERE permission_id = %s'''
```

Comment:
Instead of using a mix of sql and python, try converting this entire method into one query which grabs user_ids

Reply:
I think it's hard because there is a rather complex query needed to find direct and indirect position holders in a group

Reply:
... unless the groups.get_position_holders method is crazy and needs to be done separately

Reply:
I'm with Caleb on this -- could do this as one query but I would be duplicating what's already written in groups as a subquery